psh/test-cat.py

```diff
@@ -27,8 +25,61 @@ def assert_cat_h(p):
     psh.assert_cmd(p, 'cat -h', expected=help, result='success')
 
 
+def assert_cat_err(p):
+    fname = 'nonexistentFile'
+
+    statement = f'cat: {fname}: ENOENT'
+    psh.assert_cmd(p, f'cat {fname}', expected=statement, result='fail')
+
+    statement = rf'cat: {ROOT_TEST_DIR}: EISDIR'
+    psh.assert_cmd(p, f'cat {ROOT_TEST_DIR}', expected=statement, result='fail')
+
+
+def assert_cat_basic(p):
+    fname = ROOT_TEST_DIR + "/textFile.txt"
+    secFname = ROOT_TEST_DIR + "/textFile2.txt"
+    fspecial = ROOT_TEST_DIR + "/text!#$?@File.txt"
+
+    text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. "
+    secText = "Curabitur eu velit a sem porttitor blandit."
+    ascii = "0123456789?@ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
```

Comment:
Let's check all printable characters using `string.printable`

Reply:
So, yes, it is possible to use a larger number of characters. I have now added more. 

In theory, by using the echo command, we can write them all thanks to the `-e` argument in echo. However, it is not available on Phoenix.

Reply:
how about whitespaces? We can also test writing them.